plugin/core/rpc.py

```diff
@@ -87,13 +87,21 @@ def __init__(self, transport: Transport, settings) -> None:
     def send_request(self, request: Request, handler: 'Callable[[Optional[Any]], None]',
                      error_handler: 'Optional[Callable]' = None) -> None:
         self.request_id += 1
-        debug(' --> ' + request.method)
-        self._response_handlers[self.request_id] = (handler, error_handler)
-        self.send_payload(request.to_payload(self.request_id))
+        if self.transport is not None:
+            debug(' --> ' + request.method)
+            self._response_handlers[self.request_id] = (handler, error_handler)
+            self.send_payload(request.to_payload(self.request_id))
+        else:
+            debug(' sending ' + request.method + ' omitted')
+            if error_handler is not None:
+                error_handler()
 
     def send_notification(self, notification: Notification) -> None:
-        debug(' --> ' + notification.method)
-        self.send_payload(notification.to_payload())
+        if self.transport is not None:
+            debug(' --> ' + notification.method)
+            self.send_payload(notification.to_payload())
+        else:
+            debug(' sending ' + notification.method + ' omitted')
```

Comment:
same here.

Reply:
@rwols The console now reads

    LSP:  --> textDocument/formatting
    Failure writing to stdout
    Traceback (most recent call last):
      File "\AppData\Roaming\Sublime Text 3\Packages\LSP\plugin\core\transports.py", line 201, in write_stdin
        self.process.stdin.flush()
    OSError: [Errno 22] Invalid argument

    LSP: transport failed
    LSP: unloading session dkmv
    LSP: unable to send shutdown
    LSP: unable to send exit